src/game/protos/scene.c

```diff
@@ -154,7 +155,18 @@ int scene_unserialize(scene *s, serial *ser) {
 int scene_clone(scene *src, scene *dst) {
 
     memcpy(dst, src, sizeof(scene));
-    src->clone(src, dst);
+    if (src->clone) {
+        src->clone(src, dst);
+    }
+    return 0;
+}
+
+int scene_clone_free(scene *sc) {
+    if (sc->clone_free) {
+        sc->clone_free(sc);
+    }
+    memset(sc, 0, sizeof(scene));
+    omf_free(sc);
```

Comment:
Why do we first memset something and then free it?

Reply:
I was debugging without asan, and it was hard to tell if I was freeing it